clippy_utils/src/ty.rs

```diff
@@ -43,6 +43,15 @@ pub fn contains_ty(ty: Ty<'_>, other_ty: Ty<'_>) -> bool {
     })
 }
 
+/// Walks into `ty` and returns `true` if any inner type is any instance of the given abstract data
+/// type.`
+pub fn contains_adt(ty: Ty<'_>, adt: &AdtDef) -> bool {
```

Comment:
I think this isn't the best name or comment for this, what this is doing is checking if a type contains an instance of a different ADT type in generic form. Any idea what would be more descriptive here? I can come up with something if you are having trouble.

Reply:
Would `contains_adt_constructor` work better? It's still a little inaccurate given that it also works with non-generic types.

Reply:
Yeah I guess that makes sense.